Makefile

```diff
@@ -4,12 +4,13 @@ PY_PACKAGE := ogr
 OGR_IMAGE := ogr
 
 build: recipe.yaml
+	if ! test -x /usr/bin/ansible-bender ; then sudo dnf -y install ansible-bender ; fi
```

Comment:
I think `-y` should be removed here, as this might potentially install software on someones system, and they might want to first review what is installed.

Also, mind that by doing this, these make targets won't be usable on any other Linux distro then Fedora Linux and family (where `dnf` is used for package management). Sometimes we have contributors who are using Ubuntu or even MacOS. So having a list of dependencies in `CONTRIBUTING.md` would be more useful for them. In part, this was the reason I was arguing to create a separate target to install the required tooling, independent of `make check`.

And lastly: if `dnf` is used to install packages, probably using `rpm` to check if those packages are already installed would be semantically more correct and more resilient.

Reply:
I agree about (dnf) -y, EXCEPT that Zuul really hates being presented
with the interactive dialogue; see the results of the latest push
below, summarized here:
...
    Install  2 Packages

    Total download size: 229 k
    Installed size: 891 k
    Is this ok [y/N]: Operation aborted.
    make: *** [Makefile:20: check] Error 1

At the same time, I introduced some code to detect whether the OS is
Fedora and set (new) PACKAGE_CHECKER and PACKAGE_INSTALLER
accordingly.

I totally agree about using rpm (rpm -q --quiet) to check whether
packages are already installed, so PACKAGE_CHECKER = rpm -q --quiet
on Fedora.

Reply:
> I agree about (dnf) -y, EXCEPT that Zuul really hates being presented
> with the interactive dialogue; see the results of the latest push
> below, summarized here:
> ...
> Install 2 Packages
> 
> ```
> Total download size: 229 k
> Installed size: 891 k
> Is this ok [y/N]: Operation aborted.
> make: *** [Makefile:20: check] Error 1
> ```

Yet another argument to have a **separate** make target to install dependencies and one to run the tests.